ethers/signer.nim

```diff
@@ -73,17 +129,33 @@ method populateTransaction*(signer: Signer,
   if chainId =? transaction.chainId and chainId != await signer.getChainId():
     raiseSignerError("chain id mismatch")
 
-  var populated = transaction
+  var populated = await signer.ensureNonceSequence(transaction)
 
-  if transaction.sender.isNone:
+  if populated.sender.isNone:
     populated.sender = some(await signer.getAddress())
-  if transaction.nonce.isNone:
-    populated.nonce = some(await signer.getNonce())
-  if transaction.chainId.isNone:
+  if populated.chainId.isNone:
     populated.chainId = some(await signer.getChainId())
-  if transaction.gasPrice.isNone and (transaction.maxFee.isNone or transaction.maxPriorityFee.isNone):
+  if populated.gasPrice.isNone and (populated.maxFee.isNone or populated.maxPriorityFee.isNone):
     populated.gasPrice = some(await signer.getGasPrice())
-  if transaction.gasLimit.isNone:
-    populated.gasLimit = some(await signer.estimateGas(populated))
+  if populated.nonce.isNone:
+    populated.nonce = some(await signer.getNonce())
+  if populated.gasLimit.isNone:
+          populated.gasLimit = some(await signer.estimateGas(populated))
```

Comment:
It feels a bit weird that we're estimating gas before setting all properties, such as gasLimit and sender.
I would suggest to either do the call to `ensureNonceSequence` last, or to combine `populateTransaction` and `ensureNonceSequence` into a single method.

Reply:
Fair point. I've merged `ensureNonceSequence` into `populateTransaction`.